### ISSUE TITLE
Vagrant startup failed

### ISSUE THREAD

**Issue by Alexwijn** [progrium/dokku#1183]
I'm using v0.3.17 and cloned it to my local Windows computer. But when I tried "vagrant up" the build would fail. I'm using [cmder](http://gooseberrycreative.com/cmder/) to type in my commands.

My log:

    λ vagrant up
    Bringing machine 'dokku' up with 'virtualbox' provider...
    ==> dokku: Importing base box 'trusty'...
    ==> dokku: Matching MAC address for NAT networking...
    ==> dokku: Setting the name of the VM: dokku_dokku_1431451631797_81655
    ==> dokku: Clearing any previously set forwarded ports...
    ==> dokku: Clearing any previously set network interfaces...
    ==> dokku: Preparing network interfaces based on configuration...
           dokku: Adapter 1: nat
           dokku: Adapter 2: hostonly
    ==> dokku: Forwarding ports...
           dokku: 80 => 8080 (adapter 1)
           dokku: 22 => 2222 (adapter 1)
    ==> dokku: Running 'pre-boot' VM customizations...
    ==> dokku: Booting VM...
    ==> dokku: Waiting for machine to boot. This may take a few minutes...
           dokku: SSH address: 127.0.0.1:2222
           dokku: SSH username: vagrant
           dokku: SSH auth method: private key
           dokku: Warning: Remote connection disconnect. Retrying...
           dokku:
           dokku: Vagrant insecure key detected. Vagrant will automatically replace
           dokku: this with a newly generated keypair for better security.
           dokku:
           dokku: Inserting generated public key within guest...
           dokku: Removing insecure key from the guest if its present...
           dokku: Key inserted! Disconnecting and reconnecting using new SSH key...
    ==> dokku: Machine booted and ready!
    ==> dokku: Checking for guest additions in VM...
    ==> dokku: Setting hostname...
    ==> dokku: Configuring and enabling network interfaces...
    ==> dokku: Mounting shared folders...
           dokku: /vagrant => C:/Users/Alex/Documents/Vagrant/dokku
           dokku: /root/dokku => C:/Users/Alex/Documents/Vagrant/dokku
    ==> dokku: Running provisioner: file...
    ==> dokku: Running provisioner: shell...
           dokku: Running: inline script
    ==> dokku: stdin: is not a tty
    ==> dokku: Running provisioner: shell...
           dokku: Running: inline script
    ==> dokku: stdin: is not a tty
    ==> dokku: wget -qO /usr/local/bin/sshcommand https://raw.github.com/progrium/sshcommand/master/sshcommand
    ==> dokku: chmod +x /usr/local/bin/sshcommand
    ==> dokku: sshcommand create dokku /usr/local/bin/dokku
    ==> dokku: Adding user `dokku' ...
    ==> dokku: Adding new group `dokku' (1002) ...
    ==> dokku: Adding new user `dokku' (1002) with group `dokku' ...
    ==> dokku: Creating home directory `/home/dokku' ...
    ==> dokku: Copying files from `/etc/skel' ...
    ==> dokku: wget -qO /tmp/pluginhook_0.1.0_amd64.deb https://s3.amazonaws.com/progrium-pluginhook/pluginhook_0.1.0_amd64.deb
    ==> dokku: dpkg -i /tmp/pluginhook_0.1.0_amd64.deb
    ==> dokku: Selecting previously unselected package pluginhook.
    ==> dokku: (Reading database ... 61720 files and directories currently installed.)
    ==> dokku: Preparing to unpack .../tmp/pluginhook_0.1.0_amd64.deb ...
    ==> dokku: Unpacking pluginhook (0.1.0) ...
    ==> dokku: Setting up pluginhook (0.1.0) ...
    ==> dokku: lsmod | grep aufs || modprobe aufs || apt-get install -qq -y linux-image-extra-`uname -r` > /dev/null
    ==> dokku: modprobe: FATAL: Module aufs not found.
    ==> dokku: run-parts: executing /etc/kernel/postinst.d/apt-auto-removal
    ==> dokku:  3.13.0-48-generic
    ==> dokku:  /boot/vmlinuz-3.13.0-48-generic
    ==> dokku: run-parts: executing /etc/kernel/postinst.d/dkms
    ==> dokku:  3.13.0-48-generic
    ==> dokku:  /boot/vmlinuz-3.13.0-48-generic
    ==> dokku: run-parts: executing /etc/kernel/postinst.d/initramfs-tools
    ==> dokku:  3.13.0-48-generic
    ==> dokku:  /boot/vmlinuz-3.13.0-48-generic
    ==> dokku: update-initramfs: Generating /boot/initrd.img-3.13.0-48-generic
    ==> dokku: run-parts: executing /etc/kernel/postinst.d/update-notifier
    ==> dokku:  3.13.0-48-generic
    ==> dokku:  /boot/vmlinuz-3.13.0-48-generic
    ==> dokku: run-parts: executing /etc/kernel/postinst.d/zz-update-grub
    ==> dokku:  3.13.0-48-generic
    ==> dokku:  /boot/vmlinuz-3.13.0-48-generic
    ==> dokku: Generating grub configuration file ...
    ==> dokku: Found linux image: /boot/vmlinuz-3.13.0-48-generic
    ==> dokku: Found initrd image: /boot/initrd.img-3.13.0-48-generic
    ==> dokku: done
    ==> dokku: apt-get install -qq -y curl
    ==> dokku: egrep -i "^docker" /etc/group || groupadd docker
    ==> dokku: usermod -aG docker dokku
    ==> dokku: curl --silent https://get.docker.com/gpg | apt-key add -
    ==> dokku: OK
    ==> dokku: echo deb http://get.docker.io/ubuntu docker main > /etc/apt/sources.list.d/docker.list
    ==> dokku: apt-get update
    ==> dokku: Ign http://archive.ubuntu.com trusty InRelease
    ==> dokku: Ign http://archive.ubuntu.com trusty-updates InRelease
    ==> dokku: Hit http://archive.ubuntu.com trusty Release.gpg
    ==> dokku: Get:1 http://archive.ubuntu.com trusty-updates Release.gpg [933 B]
    ==> dokku: Hit http://archive.ubuntu.com trusty Release
    ==> dokku: Get:2 http://archive.ubuntu.com trusty-updates Release [63.5 kB]
    ==> dokku: Get:3 http://archive.ubuntu.com trusty/main Sources [1,064 kB]
    ==> dokku: Ign http://security.ubuntu.com trusty-security InRelease
    ==> dokku: Get:4 http://security.ubuntu.com trusty-security Release.gpg [933 B]
    ==> dokku: Ign http://get.docker.io docker InRelease
    ==> dokku: Get:5 http://security.ubuntu.com trusty-security Release [63.5 kB]
    ==> dokku: Get:6 http://archive.ubuntu.com trusty/universe Sources [6,399 kB]
    ==> dokku: Get:7 http://security.ubuntu.com trusty-security/main Sources [80.7 kB]
    ==> dokku: Get:8 http://security.ubuntu.com trusty-security/universe Sources [21.9 kB]
    ==> dokku: Get:9 http://get.docker.io docker Release.gpg [473 B]
    ==> dokku: Get:10 http://security.ubuntu.com trusty-security/main amd64 Packages [268 kB]
    ==> dokku: Get:11 http://security.ubuntu.com trusty-security/universe amd64 Packages [101 kB]
    ==> dokku: Hit http://archive.ubuntu.com trusty/main amd64 Packages
    ==> dokku: Hit http://archive.ubuntu.com trusty/universe amd64 Packages
    ==> dokku: Hit http://archive.ubuntu.com trusty/main Translation-en
    ==> dokku: Get:12 http://security.ubuntu.com trusty-security/main Translation-en [136 kB]
    ==> dokku: Hit http://archive.ubuntu.com trusty/universe Translation-en
    ==> dokku: Get:13 http://get.docker.io docker Release [1,525 B]
    ==> dokku: Get:14 http://archive.ubuntu.com trusty-updates/main Sources [199 kB]
    ==> dokku: Get:15 http://archive.ubuntu.com trusty-updates/universe Sources [114 kB]
    ==> dokku: Get:16 http://archive.ubuntu.com trusty-updates/main amd64 Packages [514 kB]
    ==> dokku: Get:17 http://security.ubuntu.com trusty-security/universe Translation-en [56.8 kB]
    ==> dokku: Get:18 http://archive.ubuntu.com trusty-updates/universe amd64 Packages [277 kB]
    ==> dokku: Get:19 http://archive.ubuntu.com trusty-updates/main Translation-en [244 kB]
    ==> dokku: Get:20 http://archive.ubuntu.com trusty-updates/universe Translation-en [144 kB]
    ==> dokku: Get:21 http://get.docker.io docker/main amd64 Packages [5,851 B]
    ==> dokku: Ign http://archive.ubuntu.com trusty/main Translation-en_US
    ==> dokku: Ign http://archive.ubuntu.com trusty/universe Translation-en_US
    ==> dokku: Ign http://get.docker.io docker/main Translation-en_US
    ==> dokku: Ign http://get.docker.io docker/main Translation-en
    ==> dokku: Fetched 9,756 kB in 9s (1,037 kB/s)
    ==> dokku: Reading package lists...
    ==> dokku: apt-get install -qq -y lxc-docker
    ==> dokku: Selecting previously unselected package aufs-tools.
    ==> dokku: (Reading database ... 65683 files and directories currently installed.)
    ==> dokku: Preparing to unpack .../aufs-tools_1%3a3.2+20130722-1.1_amd64.deb ...
    ==> dokku: Unpacking aufs-tools (1:3.2+20130722-1.1) ...
    ==> dokku: Selecting previously unselected package cgroup-lite.
    ==> dokku: Preparing to unpack .../cgroup-lite_1.9_all.deb ...
    ==> dokku: Unpacking cgroup-lite (1.9) ...
    ==> dokku: Selecting previously unselected package lxc-docker-1.6.1.
    ==> dokku: Preparing to unpack .../lxc-docker-1.6.1_1.6.1_amd64.deb ...
    ==> dokku: Unpacking lxc-docker-1.6.1 (1.6.1) ...
    ==> dokku: Selecting previously unselected package lxc-docker.
    ==> dokku: Preparing to unpack .../lxc-docker_1.6.1_amd64.deb ...
    ==> dokku: Unpacking lxc-docker (1.6.1) ...
    ==> dokku: Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
    ==> dokku: Processing triggers for ureadahead (0.100.0-16) ...
    ==> dokku: Setting up aufs-tools (1:3.2+20130722-1.1) ...
    ==> dokku: Setting up cgroup-lite (1.9) ...
    ==> dokku: cgroup-lite start/running
    ==> dokku: Setting up lxc-docker-1.6.1 (1.6.1) ...
    ==> dokku: docker start/running, process 14350
    ==> dokku: Processing triggers for ureadahead (0.100.0-16) ...
    ==> dokku: Setting up lxc-docker (1.6.1) ...
    ==> dokku: Processing triggers for libc-bin (2.19-0ubuntu6.6) ...
    ==> dokku: sleep 2 # give docker a moment i guess
    ==> dokku: apt-get install -qq -y help2man
    ==> dokku: Selecting previously unselected package help2man.
    ==> dokku: (Reading database ... 65804 files and directories currently installed.)
    ==> dokku: Preparing to unpack .../help2man_1.44.1_amd64.deb ...
    ==> dokku: Unpacking help2man (1.44.1) ...
    ==> dokku: Processing triggers for install-info (5.2.0.dfsg.1-2) ...
    ==> dokku: Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
    ==> dokku: Setting up help2man (1.44.1) ...
    ==> dokku: make -e stack
    ==> dokku: make[1]: Entering directory `/root/dokku'
    ==> dokku: Start building buildstep
    ==> dokku: 9bbac4d308ea579b11dc75f6a2b14226b8748ca0ae533c181ca83d47ca00e061
    ==> dokku: make[1]: Leaving directory `/root/dokku'
    ==> dokku: cp dokku /usr/local/bin/dokku
    ==> dokku: mkdir -p /var/lib/dokku/plugins
    ==> dokku: find /var/lib/dokku/plugins -mindepth 2 -maxdepth 2 -name '.core' -printf '%h\0' | xargs -0 rm -Rf
    ==> dokku: find plugins/ -mindepth 1 -maxdepth 1 -type d -printf '%f\n' | while read plugin; do \
    ==> dokku:              rm -Rf /var/lib/dokku/plugins/$plugin && \
    ==> dokku:              cp -R plugins/$plugin /var/lib/dokku/plugins && \
    ==> dokku:              touch /var/lib/dokku/plugins/$plugin/.core; \
    ==> dokku:              done
    ==> dokku: make addman
    ==> dokku: make[1]: Entering directory `/root/dokku'
    ==> dokku: mkdir -p /usr/local/share/man/man1
    ==> dokku: help2man -Nh help -v version -n "configure and get information from your dokku installation" -o /usr/local/share/man
    /man1/dokku.1 dokku
    ==> dokku: help2man: can't get `help' info from dokku
    ==> dokku: Try `--no-discard-stderr' if option outputs to stderr
    ==> dokku: make[1]:
    ==> dokku: *** [addman] Error 127
    ==> dokku: make[1]: Leaving directory `/root/dokku'
    ==> dokku: make:
    ==> dokku: *** [copyfiles] Error 2
    The SSH command responded with a non-zero exit status. Vagrant
    assumes that this means the command failed. The output for this command
    should be in the log above. Please read the output to determine what
    went wrong.

Any advice? Thanks.

**Comment by michaelshobbs** [comment]
this looks like a busted clone actually. i would destroy your vagrant machine, pull from master and try again.

**Comment by JJWTimmer** [comment]
I'm having the same problem.

I just pulled and ran vagrant up, and when reading about a corrupt clone, I tried again, so that's not it.

I'm using ConEmu itself, instead of cmder, trying now with vanilla cmd, see if that makes a difference.

**Comment by michaelshobbs** [comment]
I'm not familiar with those tools. If you're seeing the exact same error, it would be interesting to see the output of `dokku help` from inside the VM.

**Comment by JJWTimmer** [comment]
I tried `vagrant ssh` and then `dokku help` which only resulted in a vague error about cannot open file. I inspected dokku with vi, and saw windows line endings. It appeared my desktop git client did auto crlf, checking out linux line ends as windows line ends on windows, and vagrant copied the windows line endings file in the vm.

tl;dr: fix your git setup.

**Comment by michaelshobbs** [comment]
Closing then as it seems we have nothing to do here. Feel free to re-open if necessary.
